### PR TITLE
eBPF Soft IRQ latency

### PR DESCRIPTION
**Pull request by UmanShahzad**
##### Summary

This PR brings eBPF-based soft IRQ latency tracking. https://github.com/netdata/product/issues/2154

##### Component Name

ebpf.plugin

##### Test Plan

- Compile this branch
- Check that a soft IRQ latency graph is showing in the system overview section.

Tested with:

- Ubuntu 18.04.1 (Linux 4.15)
- Ubuntu 21.04 (Linux 5.11.0-31)
- CentOS 7 (Linux 3.10)

### CODE REVIEW

**Comment by UmanShahzad** [comment]
Example graph:

image

File: web/gui/dashboard_info.js [hunk 1]
```diff
@@ -960,6 +960,10 @@ netdataDashboard.context = {
         info: 'CPU softirqs in detail. At the <a href="#menu_cpu">CPUs</a> section, softirqs are analyzed per CPU core.'
     },
 
+    'system.softirq_latency': {
+        info: 'Total time spent servicing software interrupts.'
```
Comment (thiagoftsm) [inline web/gui/dashboard_info.js:964]: I did not review this PR yet, but while I am doing tests to verify that eBPF program is running on different kernels, it called my attention that we are not saying that the new charts is coming from `eBPF.plugin`, please, add a reference to the plugin here.

This is something already discussed with @hugovalente-pm and @manos-saratsis and it is one of the biggest motivations for me work in this [issue](https://github.com/netdata/netdata/issues/11439).

Thank you!

**Comment by thiagoftsm** [comment]
When I was testing on `CentOS 7.9` I got this strange values:

Screenshot_20210820_140352

The eBPF source code looks OK, but the values that we are getting when we run on `3.10.0-1160` it is not expected, because I am running VM only 30 minutes.

**Comment by UmanShahzad** [comment]
@thiagoftsm I have some candidate reasons for why that could be happening. But I'd like to know: are you using 32-bit `CentOS 7.9`?

**Comment by UmanShahzad** [comment]
@thiagoftsm I tried on CentOS 7 now myself, and the graph looks fine to me:

VirtualBox_centos3-1_22_08_2021_02_30_25

File: web/gui/dashboard_info.js (hunk 1 above)
Comment (UmanShahzad) [reply web/gui/dashboard_info.js]: Added in https://github.com/netdata/netdata/pull/11445/commits/75edc650dc3653a2c5f4197a932c113524a97902

**Comment by thiagoftsm** [comment]
> @thiagoftsm I have some candidate reasons for why that could be happening. But I'd like to know: are you using 32-bit `CentOS 7.9`?

I am not using `32-bit`,my host and VMs are 100% `64-bit`, I retest today and I had problems again, but considering you could run without problems, it can be a problem on my VM, because it is an old VM. I am glad with your results, thank you!

File: collectors/ebpf.plugin/ebpf_softirq.c [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+#define SOFTIRQ_TP_CLASS_IRQ "irq"
+static ebpf_tracepoint_t softirq_tracepoints[] = {
+    {.enabled = false, .class = SOFTIRQ_TP_CLASS_IRQ, .event = "softirq_entry"},
+    {.enabled = false, .class = SOFTIRQ_TP_CLASS_IRQ, .event = "softirq_exit"},
+    /* end */
+    {.enabled = false, .class = NULL, .event = NULL}
+};
+
+// these must be in the order defined by the kernel.
```
Comment (thiagoftsm) [inline collectors/ebpf.plugin/ebpf_softirq.c:41]: Considering that kernel is keeping this list relatively stable, I suggest to add this [link](https://elixir.bootlin.com/linux/v5.12.19/source/include/trace/events/irq.h#L13) for the comment.

File: collectors/ebpf.plugin/ebpf_softirq.c [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    return NULL;
+}
+
+static void softirq_create_charts()
+{
+    ebpf_create_chart(
+        NETDATA_EBPF_SYSTEM_GROUP,
+        "softirq_latency",
+        "Software IRQ latency",
+        "milliseconds",
```
Comment (thiagoftsm) [inline collectors/ebpf.plugin/ebpf_softirq.c:160]: We have a list of common units [here](https://github.com/netdata/netdata/blob/master/collectors/ebpf.plugin/ebpf.h#L219-L225), considering this is an "unit" used with `hard` and `soft` IRQ, I will suggest you to create a preprocessor for it.

**Comment by thiagoftsm** [comment]
@UmanShahzad overall the code is good, I loved it, I only had small requests for this. When the PR https://github.com/netdata/kernel-collector/pull/251 is merged and a new release is added for this issue, I will retest it.

File: collectors/ebpf.plugin/ebpf_softirq.c (hunk 3 above)
Comment (UmanShahzad) [reply collectors/ebpf.plugin/ebpf_softirq.c]: Addressed in https://github.com/netdata/netdata/pull/11445/commits/ba9cbddce1ebb91fccbc059a8fbdd7cc0a386424

File: collectors/ebpf.plugin/ebpf_softirq.c (hunk 2 above)
Comment (UmanShahzad) [reply collectors/ebpf.plugin/ebpf_softirq.c]: Addressed in https://github.com/netdata/netdata/pull/11445/commits/ba9cbddce1ebb91fccbc059a8fbdd7cc0a386424

**Review by vlvkobal — APPROVED** [review]

**Review by thiagoftsm — APPROVED** [review]
I tested this PR on:

| Linux Distribution | kernel version |
|-------------------|----------------|
| Slackware Current | 5.13.12  |
| Manjaro 21.1 | 5.10.56 |
| Slackware Current | 4.14.239 |
| CentOS 7.9.2009 | 3.10.0-1160 |

Everything worked as expected, except for a VM with problems, but considering that @UmanShahzad had success, we can merge this PR without problems.

**Pull request merged**